components/ui/tabcounter/src/main/java/mozilla/components/ui/tabcounter/TabCounter.kt

```diff
@@ -19,19 +19,18 @@ import mozilla.components.support.ktx.android.view.onNextGlobalLayout
 import mozilla.components.support.utils.DrawableUtils
 import java.text.NumberFormat
 
-@Suppress("LargeClass")
 open class TabCounter @JvmOverloads constructor(
     context: Context,
     attrs: AttributeSet? = null,
-    defStyle: Int = 0
+    defStyle: Int = 0,
+    private var tabCount: Int = 0
```

Comment:
Is there a reason we need this to be a property?

Reply:
>But simplify everything else and just set the count to 0 in the constructor.
You said set in constructor and i did, i get you wrong?

Reply:
Sorry, I think I made it more confusing.  I meant making a call inside the constructor to set it to 0.  Didn't mean to sound like I wanted a new property.  My bad.

Reply:
I'm completely lost :)
The `count` used in `setCountWithAnimation` is different than current `tabCount`. We need `tabCount` to get initial state.
There is two seperate tab count currently and before this pr. `tabCount` is used for `don't animate on initial state` etc. but `count` is used for getting tab count from `homefragment` etc.

If you mean something another, please send the code how it should be :(

Reply:
It could be me.  I don't know this code as well as you do.  But I was thinking keeping the `private var count: Int = 0`.  Then on line 51 just call setCount(0).  Do you see an issue with this?

Reply:
I was moved this to companion object because everytime we run TabCounter.kt, count got reset thus stays in initial state and never animating tab counter. Line 69.
I've also opened issue about this https://github.com/mozilla-mobile/fenix/issues/11822. Tab tray covers tab counter and animations never seen by user.
What do you think?

Reply:
I see.  Moving to companion object is to fix the other issue?  If we don't move to a companion object we won't fix the flickering issue?  

I am asking this because I would prefer one fix for one issue.  If we can fix the flickering without moving to companion object, I would rather merge the change without moving to companion object.  After this you can always create a new A-C issue and put up a new PR for the animations problem

Reply:
We can fix flicker without this change. Gonna roll it back.